src/scheduler/mod.rs

```diff
@@ -328,6 +340,45 @@ impl PerCoreScheduler {
 			.as_u64();
 	}
 
+	pub fn set_current_task_priority(&mut self, prio: Priority) {
+		irqsave(|| {
+			info!("Change priority of the current task");
+			self.current_task.borrow_mut().prio = prio;
+		});
+	}
+
+	pub fn set_priority(&mut self, id: TaskId, prio: Priority) -> Result<(), ()> {
+		info!("Chabge priority of task {} to priority {}", id, prio);
+
+		irqsave(|| {
+			let task = get_task_handle(id).ok_or(())?;
+
+			#[cfg(feature = "smp")]
+			if task.get_core_id() == self.core_id {
+				if self.current_task.borrow().id == task.get_id() {
+					self.current_task.borrow_mut().prio = prio;
+				} else {
+					self.ready_queue
+						.set_priority(task, prio)
+						.expect("Do not find valid task in ready queue");
+				}
+			} else {
+				warn!("Have to change the priority on another core");
```

Comment:
Is printing a warning and failing silently otherwise correct here? :thinking: 
Should it be possible in the future to change the priority on other cores?
The alternatives would be to panic or to return an error (results in a panic when unwrapping in the  API).

Reply:
Yes, I want also to change the priority on other cores. I am working on it.